Client/Application/AppDelegate.swift

```diff
@@ -359,57 +359,73 @@ class AppDelegate: UIResponder, UIApplicationDelegate, UIViewControllerRestorati
             return false
         }
 
-        if AppConstants.MOZ_FXA_DEEP_LINK_FORM_FILL {
-            // Extract optional FxA deep-linking options
-            let query = url.getQuery()
-            let host = url.host
-            
-            // FxA form filling requires a `signin` query param and host = fxa-signin
-            // Ex. firefox://fxa-signin?signin=<token>&someQuery=<data>...
-            if query["signin"] != nil && host == "fxa-signin" {
-                let fxaParams: FxALaunchParams
-                fxaParams = FxALaunchParams(query: query)
-                launchFxAFromURL(fxaParams)
-                return true
-            }
+        guard let host = url.host else {
+            log.warning("Cannot handle nil URL host")
+            return false
         }
 
-        var url: String?
-        var isPrivate: Bool = false
-        
-        for item in (components.queryItems ?? []) as [URLQueryItem] {
-            switch item.name {
-            case "url":
-                url = item.value
-            case "private":
-                isPrivate = NSString(string: item.value ?? "false").boolValue
-            case "deep-link":
-                guard let value = item.value else { break }
-                Router.shared.routeURL(value)
-                return true
-            default: ()
+        switch host {
+        case "open-url":
+            var url: String?
+            var isPrivate: Bool = false
+            for item in (components.queryItems ?? []) as [URLQueryItem] {
```

Comment:
The `fxa-signin` case below uses `url.getQuery()` .. isn't that nicer? Then you can just do:

```
if let url = query["url"], let private = query["private"] ?? false {
   ...
}
```